### ISSUE TITLE
Don't relay devices which aren't e2e-capable to other users

### ISSUE THREAD

**Issue by ara4n** [matrix-org/synapse#1977]
Currently, every time someone logs in, we create a device for them (to help them manage access_tokens, as well as for E2E crypto purposes).  This can cause services like matrix-ircd however to accumulate a tonne of devices, as every time someone signs into the ircd, the ircd validates their u/p and logs them in as a new device.  This is a problem in its own right (perhaps the ircd could cache access_tokens for mxids it's seen before), but the least we can do is to not spam federation with device_list updates for devices which aren't e2e capable.

**Comment by richvdh** [comment]
(the receiving end then explodes, because of https://github.com/matrix-org/synapse/issues/1962)

**Comment by reivilibre** [comment]
I don't think this issue is valid (anymore?); devices don't need to support E2EE in order for them to be useful targets for communication over federation — e.g. Matrix RTC uses to-device messages but shouldn't necessarily depend on E2EE.
